lib/shoryuken/polling.rb

```diff
@@ -0,0 +1,88 @@
+module Shoryuken
+  module Polling
+    QueueConfiguration = Struct.new(:name, :options) do
+      def hash
+        name.hash
+      end
+
+      def ==(other)
```

Comment:
This is complex and confusing. Don't compare QueueConfigurations to Strings. Use queue_configuration.name == instead.